src/safe-nonce/safe-nonce.js

```diff
@@ -0,0 +1,16 @@
+htmx.defineExtension('safe-nonce', {
+  transformResponse: function(text, xhr, elt) {
+    if (htmx.config.inlineScriptNonce) {
+      htmx.config.inlineScriptNonce = ''
+    }
+    if (!htmx.config.refreshOnHistoryMiss) { 
+      htmx.config.refreshOnHistoryMiss = true
+    }
+    const nonce = xhr.getResponseHeader('HX-Nonce')
+    if (htmx.config.safeInlineScriptNonce && nonce) {
+      const escapedRegex = new RegExp(`nonce="${nonce.replace(/[\\\[\]\/^*.+?$(){}'#:!=|]/g, '\\$&')}"`, 'g')
```

Comment:
Here we need to do a straight text find replace of the AJAX request supplied nonce with the page load nonce stored in config.  But using .replace is regex based and we need to sanitize the passed in regex before generating a replace regex so this line escapes all regex special characters in the search term so it is safe